boards: lpcxpresso54114: Enable SEGGER support by default for M0

galak opened the pull request:
The lpcxpresso54114_m0 tends to be the secondary core so doesn't
typically have a UART setup for it.  We had SEGGER support enabled by
default previously, but the new Kconfig option CONFIG_USE_SEGGER_RTT
needs to be set as well now.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

galak committed:
boards: lpcxpresso54114: Enable SEGGER support by default for M0

The lpcxpresso54114_m0 tends to be the secondary core so doesn't
typically have a UART setup for it.  We had SEGGER support enabled by
default previously, but the new Kconfig option CONFIG_USE_SEGGER_RTT
needs to be set as well now.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

MaureenHelm commented on boards/arm/lpcxpresso54114/lpcxpresso54114_m0_defconfig:
```diff
@@ -9,6 +9,7 @@ CONFIG_SOC_LPC54114_M0=y
 CONFIG_BOARD_LPCXPRESSO54114_M0=y
 CONFIG_CONSOLE=y
+CONFIG_USE_SEGGER_RTT=y
 CONFIG_RTT_CONSOLE=y
```
Can remove `CONFIG_RTT_CONSOLE=y`, since it gets selected by `CONFIG_USE_SEGGER_RTT`

MaureenHelm reviewed (changes requested):
While you're at it, can you please make the same change in hexiwear_kw40z?

galak committed:
boards: hexiwear_kw40z: Enable SEGGER support by default

We had SEGGER support enabled by default previously, but the new
Kconfig option CONFIG_USE_SEGGER_RTT needs to be set as well now.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>
